instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/internal/JdbcConnectionUrlParser.java

```diff
@@ -831,6 +831,81 @@ DbInfo.Builder doParse(String jdbcUrl, DbInfo.Builder builder) {
 
       return MODIFIED_URL_LIKE.doParse(jdbcUrl, builder);
     }
+  },
+  INFORMIX_SQLI("informix-sqli") {
+    @Override
+    DbInfo.Builder doParse(String jdbcUrl, DbInfo.Builder builder) {
+      builder = MODIFIED_URL_LIKE.doParse(jdbcUrl, builder);
+      builder = INFORMIX.doParse(jdbcUrl, builder);
+
+      int hostIndex = jdbcUrl.indexOf("://");
+      if (hostIndex == -1) {
+        return builder;
+      }
+
+      int dbNameStartIndex = jdbcUrl.indexOf('/', hostIndex + 3);
+      if (dbNameStartIndex == -1) {
+        return builder;
+      }
+      int dbNameEndIndex = jdbcUrl.indexOf(':', dbNameStartIndex);
+      if (dbNameEndIndex == -1) {
+        dbNameEndIndex = jdbcUrl.length();
+      }
+      String name = jdbcUrl.substring(dbNameStartIndex + 1, dbNameEndIndex);
+      if (name.isEmpty()) {
+        builder.name(null);
+      } else {
+        builder.name(name);
+      }
+
+      return builder;
+    }
+  },
+
+  INFORMIX_DIRECT("informix-direct") {
+    private static final String DEFAULT_HOST = "infxhost";
+
+    @Override
+    DbInfo.Builder doParse(String jdbcUrl, DbInfo.Builder builder) {
+      builder = MODIFIED_URL_LIKE.doParse(jdbcUrl, builder);
+      builder = INFORMIX.doParse(jdbcUrl, builder);
+      builder.host(DEFAULT_HOST);
```

Comment:
To me this is strange. Looking at https://www.ibm.com/docs/en/informix-servers/12.10?topic=method-format-database-urls before `informix-direct` it says 

> For connections on the database server, use the following format:

Does this mean that `informix-direct` is used when the server is connecting to somewhere? If that is the case should we bother with it at all? Considering that it is difficult to find info about `informix-direct` it might be the best option. There doesn't seem to be a host and port for `informix-direct`, if we are going to parse this then imo we should not fill these at all (could skip `MODIFIED_URL_LIKE.doParse` and `INFORMIX.doParse` calls).